modules/discretedata/connectivity/structuredgrid.h

```diff
@@ -0,0 +1,110 @@
+/*********************************************************************************
+*
+* Inviwo - Interactive Visualization Workshop
+*
+* Copyright (c) 2012-2018 Inviwo Foundation
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*
+* 1. Redistributions of source code must retain the above copyright notice, this
+* list of conditions and the following disclaimer.
+* 2. Redistributions in binary form must reproduce the above copyright notice,
+* this list of conditions and the following disclaimer in the documentation
+* and/or other materials provided with the distribution.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+* ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+* DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+* ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+* (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+* LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+* ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*
+*********************************************************************************/
+
+#pragma once
+
+#include "discretedata/connectivity/connectivity.h"
+#include "discretedata/util.h"
+
+namespace inviwo {
+namespace dd {
+
+class ElementIterator;
+
+/** \class StructuredGrid
+    \brief A curvilinear grid in nD
+
+    @author Anke Friederici and Tino Weinkauf
+*/
+class IVW_MODULE_DISCRETEDATA_API StructuredGrid : public Connectivity {
```

Comment:
We had a look into it, but mainly two reasons against it:
- We find the model, world transforms more confusing than anything else. Though I guess it might be beneficial to have them, but did not yet find a scenario where I need them.
- It is possible to have a 1D structured grid (a line) in a 3D space. I think that can not be represented by SpatialData?

I'm very open to integrate them if there is a benefit. I don't know where they are used so far.
Any thoughts?

Reply:
Also, the grid so far is independent of the point dimension. So this is probably more relevant for the DataSet - might actually be a good place to save the transformations.

Reply:
Was also thinking that it might be more suitable in the DataSet. 
The SpatialData is used to place the data in world space. I guess the vertices must be given in world space right now?

Reply:
The spatial model could be implements as a spatial position channel if needed.